sdk/servicebus/service-bus/src/track2/serviceBusReceiverClient.ts

```diff
@@ -0,0 +1,526 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { ServiceBusMessage, ReceiveMode } from "../serviceBusMessage";
+import { ClientEntityContext } from "../clientEntityContext";
+import { generate_uuid } from "rhea-promise";
+import { ClientType } from "../client";
+import { InternalSessionReceiver, InternalReceiver } from "../internalReceivers";
+import {
+  MessageHandlers,
+  MessageIterator,
+  Session,
+  ContextType,
+  ContextWithSettlement,
+  QueueAuth,
+  SubscriptionAuth,
+  isSession,
+  MessageAndContext,
+  Message,
+  ReceiveBatchOptions,
+  IterateMessagesOptions,
+  StreamMessagesOptions
+} from "./models";
+import { createConnectionContext, convertToInternalReceiveMode } from "./constructorHelpers";
+import { RuleDescription, CorrelationFilter } from "../core/managementClient";
+import { ServiceBusClientOptions } from "../old/serviceBusClient";
+import { ConnectionContext } from "../connectionContext";
+
+/**
+ *A receiver client that handles sessions, including renewing the session lock.
+ */
+// TODO: could extend NonSessionReceiverClient...?
```

Comment:
This seems reasonable unless you want to really tailor the docs differently based on if sessions are used or not.

Reply:
I actually found one particular case (and perhaps there will be more) where I did that (peek() locks the session, while peek() for non-session-receivers does not lock anything). Comment removed.